### PR TITLE
Changing `TransactionV1` structure. From now on a `TransactionV1` con…

### PR DESCRIPTION
**Pull request by zajko**
…sists of `hash`, `payload` and `approvals`. `payload` is a merge of `header` and `body` concepts from before. `body` is not represented in a type-constrained way, instead we use an amorphic `fields` of type `BTreeMap<u16, Bytes>` which is more resistant to future changes.

### CODE REVIEW

File: internal_types/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+edition = "2021"
+description = "Types shared by internal modules of the node. Not intented to be used outside of the node or published as a crate"
+readme = "README.md"
+homepage = "https://casperlabs.io"
+repository = "https://github.com/CasperLabs/casper-node/tree/master/execution_engine_types"
+license = "Apache-2.0"
+
+[dependencies]
+casper-execution-engine = { version = "7.0.1", path = "../execution_engine" }
+casper-types = { version = "5.0.0", path = "../types", features = ["datasize", "json-schema", "std", "std-fs-io"] }
```
Comment (gRoussac) [inline internal_types/Cargo.toml]: As discussed please try without "std-fs-io"

File: internal_types/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+rand = { version = "0.8.3", default-features = false, features = ["small_rng"] }
+rand_pcg = { version = "0.3.0", optional = true }
+once_cell = { version = "1.5.2", optional = true }
+datasize = { version = "0.2.11", optional = true }
+serde = { version = "1", features = ["derive", "rc"] }
+
+[features]
+std = ["once_cell"]
+testing = ["rand/default", "rand_pcg"]
+gens = ["testing"]
```
Comment (gRoussac) [inline internal_types/Cargo.toml]: I believe "gens" can go and in [tests/Cargo.toml](https://github.com/casper-network/casper-node/pull/4890/files#diff-9218d05dd263a533d0d9c4ebe4224162c35efdb260768ded5cfbc62ffcba1c4b)

File: internal_types/Cargo.toml (hunk 1 above)
Comment (zajko) [reply internal_types/Cargo.toml]: :+1:

File: types/src/transaction/transaction_v1.rs [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
-use alloc::{collections::BTreeSet, vec::Vec};
-use core::{
-    cmp,
-    fmt::{self, Debug, Display, Formatter},
-    hash,
+use crate::{
+    bytesrepr::{self, Error, FromBytes, ToBytes},
+    crypto,
 };
+#[cfg(any(feature = "testing", test))]
```
Comment (gRoussac) [inline types/src/transaction/transaction_v1.rs]:
```suggestion
#[cfg(any(all(feature = "std", feature = "testing"), test))]
```
I believe same as random_auction

File: types/src/transaction/transaction_v1.rs [hunk 4]
```diff
@@ -1,20 +1,32 @@
...
+use crate::{
+    testing::TestRng, AUCTION_LANE_ID, INSTALL_UPGRADE_LANE_ID, LARGE_WASM_LANE_ID, MINT_LANE_ID,
+};
+#[cfg(any(feature = "std", test, feature = "testing"))]
+use alloc::collections::BTreeMap;
+use alloc::{collections::BTreeSet, vec::Vec};
+use errors_v1::FieldDeserializationError;
+#[cfg(any(feature = "testing", test))]
```
Comment (gRoussac) [inline types/src/transaction/transaction_v1.rs]:
```suggestion
#[cfg(any(all(feature = "std", feature = "testing"), test))]
```

File: types/src/transaction/transaction_v1.rs (hunk 3 above)
Comment (zajko) [reply types/src/transaction/transaction_v1.rs]: I really don't get it why we add "std" to code that is "std" agnostic, but OK I'll make it the same for the sake of consistency.

Comment (gRoussac) [reply types/src/transaction/transaction_v1.rs]: it's just cargo lint and fmt that complains because random_auction / random_install_upgrade is flag std

**Review by EdHastingsCasperAssociation — APPROVED** [review]
i'd like to get this merged...it seems basically solid, and if there remains some fit & finish we can deal with it via bug report

File: types/src/transaction.rs [hunk 5]
```diff
@@ -194,15 +186,15 @@ impl Transaction {
...
     }
-
-    /// Returns the header.
-    pub fn header(&self) -> TransactionHeader {
-        match self {
-            Transaction::Deploy(deploy) => TransactionHeader::Deploy(deploy.header().clone()),
-            Transaction::V1(transaction) => TransactionHeader::V1(transaction.header().clone()),
+    /*
+        /// Returns the header.
+        pub fn header(&self) -> TransactionHeader {
```
Comment (gRoussac) [inline types/src/transaction.rs]: I guess you can delete that commented code

**Pull request merged**